.github/workflows/build-and-test-snap.yml

```diff
@@ -0,0 +1,108 @@
+name: Snap Build and Test
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main, chardev ]
+  # Allow manual trigger
+  workflow_dispatch:
+
+jobs:
+  build-arm64:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: 
+          - arm64
+    steps:
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: ${{ matrix.platform }}
+
+      - name: Build snap for arm64
+        uses: diddlesnaps/snapcraft-multiarch-action@v1
+        id: build
+        with:
+          architecture: ${{ matrix.platform }}
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: matter-pi-gpio-commander_${{ github.run_number}}_${{ matrix.platform }}.snap
+          path: ${{ steps.build.outputs.snap }}
+
+  build-amd64:
+    outputs:
+      snap: ${{ steps.snapcraft.outputs.snap }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform:
+          - amd64
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Build Snap
+        uses: snapcore/action-build@v1
+        id: snapcraft
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: matter-pi-gpio-commander_${{ github.run_number}}_${{ matrix.platform }}.snap
+          path: ${{ steps.snapcraft.outputs.snap }}
+          if-no-files-found: error
+
+  test-amd64:
+    needs: build-amd64
+    runs-on: ubuntu-latest
```

Comment:
We don't need the `amd64` as a build artifact since it only makes sense for the tests on Github.

The two steps can be merged.

Reply:
Okay, done, merged!